js/saveState.js

```diff
@@ -0,0 +1,42 @@
+import _ from "lodash";
+
+import { COOKIE_LAST_STATE, COOKIE_OMIT_FROM_STATE } from "./constants/AppConstants";
+
+// store the cookie here in case we're testing
+let lastCookie = "";
+
+function cookiesToMap (cookieStr) {
+  return _.chain(cookieStr.split(/;\s*/))
+          .map((keyValStr) => keyValStr.split(/=/))
+          .fromPairs()
+          .value();
+}
+
+export function restoreState() {
+	let cookie = lastCookie;
+	if (typeof document !== 'undefined') {
+    cookie = document.cookie;
+	}
+  const stateEncoded = cookiesToMap(cookie)[COOKIE_LAST_STATE];
+  if (stateEncoded) {
+    const stateJson = decodeURIComponent(stateEncoded);
+    return JSON.parse(stateJson);
+  } else {
+    return {};
+  }
+}
+
+export function saveState(state) {
+	const stateToSave = _.omit(state, COOKIE_OMIT_FROM_STATE);
+	const stateJson = JSON.stringify(stateToSave);
+	const stateEncoded = encodeURIComponent(stateJson);
+	const stateCookie = COOKIE_LAST_STATE + "=" + stateEncoded;
```

Comment:
Ok, I remember that now.  In the past I've Base64 the content of cookie, which made it pretty easy to decrypt.  Looks like there are some methods that are included https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding